	public void forwardEvents(final List<Event> events) {
		CompletableFuture.completedFuture(events)
				// Send the events to Elasticsearch
				.thenAcceptAsync(this::sendEvents, executor)
				.exceptionally(e -> {
					LOG.error("Unexpected exception during task completion: " + e.getMessage(), e);
					if (e.getCause() instanceof ConnectionPoolShutdownException) {
						ExceptionUtils.handle(getClass(), (ConnectionPoolShutdownException) e.getCause(), events);
					}
					return null;
				});
	}

